File: Day144_2025/Q144_GFG.java
class Solution {
    public static int sumSubstrings(String s) {
        long sum = 0;
        long mf = 1;
        int mod = 1000000007; // Use modulus if required by the problem

        for (int i = s.length() - 1; i >= 0; i--) {
            int digit = s.charAt(i) - '0';
            sum = (sum + digit * (i + 1L) * mf) % mod;
            mf = (mf * 10 + 1) % mod;
        }

        return (int) sum;
    }
}

File: Day144_2025/Q144_GFG.py
class Solution:
    def sumSubstrings(self, s: str) -> int:
        sum_ = 0
        mf = 1
        mod = 10**9 + 7  # Use modulus if required

        for i in range(len(s) - 1, -1, -1):
            digit = int(s[i])
            sum_ = (sum_ + digit * (i + 1) * mf) % mod
            mf = (mf * 10 + 1) % mod

        return sum_
